lease/lessor.go

```diff
@@ -241,28 +260,37 @@ func (le *lessor) Recover(b backend.Backend, dr DeleteableRange) {
 	le.initAndRecover()
 }
 
+func (le *lessor) ExpiredLeasesC() <-chan []*Lease {
+	le.mu.Lock()
+	defer le.mu.Unlock()
+
+	return le.expiredC
+}
+
 func (le *lessor) runLoop() {
 	// TODO: stop runLoop
 	for {
 		le.mu.Lock()
 		if le.primary {
-			le.revokeExpiredLeases(le.findExpiredLeases())
+			ls := le.findExpiredLeases()
+			if len(ls) != 0 {
+				select {
+				case le.expiredC <- ls:
```

Comment:
would it be safe to do this outside the lock? Keeping the `expiredC <- ls` under the lock might lead to deadlock if a Promote/Demote shows up.